packages/story-editor/src/elements/media/utils/useProxyUrl.js

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback } from '@web-stories-wp/react';
+import { useFeature } from 'flagged';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../../../app';
+
+function useProxyUrl() {
+  const {
+    actions: { getProxyUrl: _getProxyUrl },
+  } = useAPI();
+  const enableCORSProxy = useFeature('enableCORSProxy');
+  const getProxiedUrl = useCallback(
+    (resource, src) => {
+      const { isExternal, id } = resource;
+      if (enableCORSProxy && !id && isExternal) {
+        return _getProxyUrl(src);
+      }
+      return src;
+    },
```

Comment:
Why the need for the second `src` param here? Why not just use `resource.src`

Reply:
It is not always src. 
See 
- [here](https://github.com/google/web-stories-wp/pull/9209/files#diff-2161e2f743055228d708a9c0e86c69928c3f13b6e9dd639789ed445f2886aa99R62)
- [here](https://github.com/google/web-stories-wp/pull/9209/files#diff-064bd410c01c910b4661dd266a5a4ce1c78991474dc2703cebd7f39d07e8e2c0R36)
- [here](https://github.com/google/web-stories-wp/pull/9209/files#diff-e4dcaa87660805a9cf2c6bacc18ee22b757f0dd6f1bbfd5ca4747591937f7e36R82)

Basically, images have source sets, so we need to proxy those as well.